src/device-state/current-state.ts

```diff
@@ -130,100 +140,119 @@ const getStateDiff = (): DeviceStatus => {
+			...conf
+		} = await config.getMany([
+			'deviceApiKey',
+			'apiTimeout',
+			'apiEndpoint',
+			'uuid',
+			'localMode',
+			'appUpdatePollInterval',
+			'hardwareMetrics',
+		]);
+
 		reportPending = true;
-		try {
-			const currentDeviceState = await deviceState.getStatus();
-			// If hardwareMetrics is false, send null patch for system metrics to cloud API
-			const info = {
-				...((await config.get('hardwareMetrics'))
-					? await sysInfo.getSystemMetrics()
-					: {
-							cpu_usage: null,
-							memory_usage: null,
-							memory_total: null,
-							storage_usage: null,
-							storage_total: null,
-							storage_block_device: null,
-							cpu_temp: null,
-							cpu_id: null,
-					  }),
-				...(await sysInfo.getSystemChecks()),
-			};
-
-			stateForReport.local = {
-				...stateForReport.local,
-				...currentDeviceState.local,
-				...info,
-			};
-			stateForReport.dependent = {
-				...stateForReport.dependent,
-				...currentDeviceState.dependent,
-			};
-
-			// Report current state
-			await report();
-			// Finishing pending report
-			reportPending = false;
-		} catch (e) {
-			eventTracker.track('Device state report failure', { error: e });
-			// We use the poll interval as the upper limit of
-			// the exponential backoff
-			const maxDelay = await config.get('appUpdatePollInterval');
-			const delay = Math.min(
-				2 ** stateReportErrors * MINIMUM_BACKOFF_DELAY,
-				maxDelay,
-			);
-
-			++stateReportErrors;
-			await Bluebird.delay(delay);
-			reportCurrentState();
-		}
+		const currentDeviceState = await deviceState.getStatus();
+		// If hardwareMetrics is false, send null patch for system metrics to cloud API
+		const info = {
+			...(hardwareMetrics
+				? await sysInfo.getSystemMetrics()
+				: {
+						cpu_usage: null,
+						memory_usage: null,
+						memory_total: null,
+						storage_usage: null,
+						storage_total: null,
+						storage_block_device: null,
+						cpu_temp: null,
+						cpu_id: null,
+				  }),
+			...(await sysInfo.getSystemChecks()),
+		};
+
+		stateForReport.local = {
+			...stateForReport.local,
+			...currentDeviceState.local,
+			...info,
+		};
+		stateForReport.dependent = {
+			...stateForReport.dependent,
+			...currentDeviceState.dependent,
+		};
+
+		const stateDiff = getStateDiff();
+
+		// report state diff
+		throttledReport(
+			{ stateDiff, conf },
+			() => {
+				// If the patch succeeds update lastReport and reset the error counter
+				_.assign(lastReportedState.local, stateDiff.local);
+				_.assign(lastReportedState.dependent, stateDiff.dependent);
+				stateReportErrors = 0;
+
+				reportPending = false;
+			},
+			(e) => {
+				if (e instanceof StatusError) {
+					// We don't want these errors to be classed as a report error, as this will cause
+					// the watchdog to kill the supervisor - and killing the supervisor will
+					// not help in this situation
+					log.error(
+						`Non-200 response from the API! Status code: ${e.statusCode} - message:`,
```

Comment:
When this error is caught it never says that the device state report failed. On Line 249 the eventTracker is passed a similar message but it doesn't go to the journal logs.

```suggestion
						`Device state report failure! Status code: ${e.statusCode} - message:`,
```

I don't think mentioning Non-200 response from the API explains what's happening.

Reply:
I've just realized the eventTracker is logged to journal but it wasn't triggered because I was testing statusCodes. The eventTracker is called when the error is not of StatusError type